core/validatorapi/router.go

```diff
@@ -682,92 +677,88 @@ func getProposeBlockParams(params map[string]string, query url.Values) (uint64,
 	return slot, randao, graffiti, err
 }
 
-// createProposeBlockResponse constructs proposeBlockResponse object for given block.
-func createProposeBlockResponse(block *eth2api.VersionedProposal) (any, error) {
-	switch block.Version {
+// createProposeBlockResponse constructs proposeBlockV3Response object for given block.
+func createProposeBlockResponse(proposal *eth2api.VersionedProposal) (any, error) {
+	var (
+		version   string
+		blockData any
+	)
+
+	switch proposal.Version {
 	case eth2spec.DataVersionPhase0:
-		if block.Phase0 == nil {
+		if proposal.Blinded {
+			return 0, errors.New("invalid blinded block")
```

Comment:
Nit: why returning 0 here? Could we return `nil` instead?

Reply:
Also, could this function get rid of the `any`  return type at all?

It is now encapsulated in `proposeBlockV3Response`, so we could return that instead.

Reply:
Perfect finding. Fixed.